cfme/tests/infrastructure/test_provisioning.py

```diff
@@ -204,5 +231,29 @@ def verify():
                 in normalize_text(mail["subject"]),
                 smtp_test.get_emails())) == len(vm_names)
         )
-
     wait_for(verify, message="email receive check", delay=5)
+
+
+@pytest.mark.parametrize("approve", [True, False], ids=["approved", "denied"])
+def test_provision_approval_msg(
```

Comment:
you use setup_provider in every test would it be better to use setup_provider once in pytestmark ?